chainer/functions/activation/rrelu.py

```diff
@@ -0,0 +1,123 @@
+import chainer
+from chainer import cuda
+from chainer import function_node
+from chainer.utils import type_check
+import numpy as np
+
+
+def _kern():
+    return cuda.elementwise(
+        'T cond, T x, T slope', 'T y',
+        'y = cond >= 0 ? x : (T)(slope * x)', 'rrelu')
+
+
+class RReLU(function_node.FunctionNode):
+    """Randomized Leaky rectifier unit."""
+
+    def __init__(self, lower=1. / 8, upper=1. / 3):
+        # lower and upper must be [0, 1) and lower <= upper
+        assert 0 <= lower < 1
+        assert 0 <= upper < 1
+        assert lower < upper
+        self.lower = lower
+        self.upper = upper
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+        x_type, = in_types
+        type_check.expect(x_type.dtype.kind == 'f')
+
+    def forward_cpu(self, x):
+        if hasattr(self, 'r'):
+            y = np.where(x[0] >= 0, x[0], x[0] * self.r)
+        else:
+            if chainer.config.train:
+                self.r = np.random.uniform(
+                    self.lower, self.upper, x[0].shape).astype(x[0].dtype)
+            else:
+                self.r = np.empty(x[0].shape).astype(x[0].dtype)
+                self.r.fill((self.lower + self.upper) / 2)
+            y = np.where(x[0] >= 0, x[0], x[0] * self.r)
+        self.retain_outputs((0,))
+        return y,
+
+    def forward_gpu(self, x):
+        xp = cuda.cupy
+        if hasattr(self, 'r'):
+            y = _kern()(x[0], x[0], self.r.astype(x[0].dtype))
+        else:
+            if chainer.config.train:
+                self.r = xp.random.uniform(
+                    self.lower, self.upper, x[0].shape).astype(x[0].dtype)
+            else:
+                self.r = xp.empty(x[0].shape).astype(x[0].dtype)
+                self.r.fill((self.lower + self.upper) / 2)
+            y = _kern()(x[0], x[0], self.r)
+        self.retain_inputs(())
+        self.retain_outputs((0,))
+        return y,
+
+    def backward(self, indexes, grad_outputs):
+        x = None
+        y = self.get_retained_outputs()[0].data
+        return _RReLUGrad(x, y, self.r).apply(grad_outputs)
+
+
+class _RReLUGrad(function_node.FunctionNode):
+    def __init__(self, x, y, r):
+        self.r = r
+        self.x = x
+        self.y = y
+
+    def forward_cpu(self, inputs):
+        gy, = inputs
+        gy = gy.copy()
```

Comment:
It seems this copy is not needed?

Reply:
Maybe, it is not needed. I referenced leaky_relu.py in older version. I modified.